An error occurred on Paper 1.13.2

I'm not sure if it is my cause. If yes, please tell me how to do it. I don't know why there are 1.17.1 errors on Paper 1.13.2.

I used the shade of 3.3.0-SNAPSHOT, and the plug-in itself is java 8.
```
[20:09:56 ERROR]: Fatal error trying to convert PartySystem v0.3:cn/yistars/party/bukkit/anvilgui/version/Wrapper1_17_R1.class
>java.lang.IllegalArgumentException: Unsupported class file major version 60
at org.objectweb.asm.ClassReader.<init>(ClassReader.java:195) ~[patched_1.13.2.jar:git-Paper-655]
at org.objectweb.asm.ClassReader.<init>(ClassReader.java:176) ~[patched_1.13.2.jar:git-Paper-655]
at org.objectweb.asm.ClassReader.<init>(ClassReader.java:162) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.craftbukkit.v1_13_R2.util.Commodore.convert(Commodore.java:170) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.craftbukkit.v1_13_R2.util.CraftMagicNumbers.processClass(CraftMagicNumbers.java:238) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:113) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.plugin.java.JavaPluginLoader.getClassByName(JavaPluginLoader.java:199) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:97) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:86) ~[patched_1.13.2.jar:git-Paper-655]
at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_281]
at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_281]
at cn.yistars.party.bukkit.anvilgui.version.VersionMatcher.<init>(VersionMatcher.java:23) ~[?:?]
at cn.yistars.party.bukkit.anvilgui.AnvilGUI.<clinit>(AnvilGUI.java:35) ~[?:?]
at cn.yistars.party.bukkit.anvilgui.AnvilGUI$Builder.open(AnvilGUI.java:491) ~[?:?]
at cn.yistars.party.bukkit.gui.Anvil.CreateAnvil(Anvil.java:29) ~[?:?]
at cn.yistars.party.bukkit.command.PartyCommand.onCommand(PartyCommand.java:33) ~[?:?]
at org.bukkit.command.PluginCommand.execute(PluginCommand.java:46) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:151) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.craftbukkit.v1_13_R2.CraftServer.dispatchCommand(CraftServer.java:734) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.PlayerConnection.handleCommand(PlayerConnection.java:1783) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1591) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:45) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:5) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:15) ~[patched_1.13.2.jar:git-Paper-655]
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_281]
at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_281]
at net.minecraft.server.v1_13_R2.SystemUtils.a(SystemUtils.java:108) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:1018) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:439) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:940) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:837) ~[patched_1.13.2.jar:git-Paper-655]
at java.lang.Thread.run(Unknown Source) [?:1.8.0_281]
[20:09:56 ERROR]: null
>org.bukkit.command.CommandException: Unhandled exception executing command 'partysystem:party' in plugin PartySystem v0.3
at org.bukkit.command.PluginCommand.execute(PluginCommand.java:48) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:151) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.craftbukkit.v1_13_R2.CraftServer.dispatchCommand(CraftServer.java:734) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.PlayerConnection.handleCommand(PlayerConnection.java:1783) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1591) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:45) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.PacketPlayInChat.a(PacketPlayInChat.java:5) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.lambda$ensureMainThread$0(PlayerConnectionUtils.java:15) ~[patched_1.13.2.jar:git-Paper-655]
at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:1.8.0_281]
at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:1.8.0_281]
at net.minecraft.server.v1_13_R2.SystemUtils.a(SystemUtils.java:108) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:1018) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:439) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:940) ~[patched_1.13.2.jar:git-Paper-655]
at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:837) ~[patched_1.13.2.jar:git-Paper-655]
at java.lang.Thread.run(Unknown Source) [?:1.8.0_281]
>Caused by: java.lang.UnsupportedClassVersionError: cn/yistars/party/bukkit/anvilgui/version/Wrapper1_17_R1 has been compiled by a more recent version of the Java Runtime (class file version 60.0), this version of the Java Runtime only recognizes class file versions up to 52.0
at java.lang.ClassLoader.defineClass1(Native Method) ~[?:1.8.0_281]
at java.lang.ClassLoader.defineClass(Unknown Source) ~[?:1.8.0_281]
at java.security.SecureClassLoader.defineClass(Unknown Source) ~[?:1.8.0_281]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:136) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.plugin.java.JavaPluginLoader.getClassByName(JavaPluginLoader.java:199) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:97) ~[patched_1.13.2.jar:git-Paper-655]
at org.bukkit.plugin.java.PluginClassLoader.findClass(PluginClassLoader.java:86) ~[patched_1.13.2.jar:git-Paper-655]
at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_281]
at java.lang.ClassLoader.loadClass(Unknown Source) ~[?:1.8.0_281]
at cn.yistars.party.bukkit.anvilgui.version.VersionMatcher.<init>(VersionMatcher.java:23) ~[?:?]
at cn.yistars.party.bukkit.anvilgui.AnvilGUI.<clinit>(AnvilGUI.java:35) ~[?:?]
at cn.yistars.party.bukkit.anvilgui.AnvilGUI$Builder.open(AnvilGUI.java:491) ~[?:?]
at cn.yistars.party.bukkit.gui.Anvil.CreateAnvil(Anvil.java:29) ~[?:?]
at cn.yistars.party.bukkit.command.PartyCommand.onCommand(PartyCommand.java:33) ~[?:?]
at org.bukkit.command.PluginCommand.execute(PluginCommand.java:46) ~[patched_1.13.2.jar:git-Paper-655]
... 15 more
```

--- a/1_10_R1/pom.xml
+++ b/1_10_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_10_R1</artifactId>

--- a/1_11_R1/pom.xml
+++ b/1_11_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_11_R1</artifactId>

--- a/1_12_R1/pom.xml
+++ b/1_12_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_12_R1</artifactId>

--- a/1_13_R1/pom.xml
+++ b/1_13_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_13_R1</artifactId>

--- a/1_13_R2/pom.xml
+++ b/1_13_R2/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_13_R2</artifactId>

--- a/1_14_4_R1/pom.xml
+++ b/1_14_4_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <artifactId>anvilgui-parent</artifactId>
         <groupId>net.wesjd</groupId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_14_4_R1</artifactId>

--- a/1_14_R1/pom.xml
+++ b/1_14_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_14_R1</artifactId>

--- a/1_15_R1/pom.xml
+++ b/1_15_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_15_R1</artifactId>

--- a/1_16_R1/pom.xml
+++ b/1_16_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_16_R1</artifactId>

--- a/1_16_R2/pom.xml
+++ b/1_16_R2/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_16_R2</artifactId>

--- a/1_16_R3/pom.xml
+++ b/1_16_R3/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_16_R3</artifactId>

--- a/1_17_R1/pom.xml
+++ b/1_17_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_17_R1</artifactId>

--- a/1_7_R4/pom.xml
+++ b/1_7_R4/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_7_R4</artifactId>

--- a/1_8_R1/pom.xml
+++ b/1_8_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_8_R1</artifactId>

--- a/1_8_R2/pom.xml
+++ b/1_8_R2/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_8_R2</artifactId>

--- a/1_8_R3/pom.xml
+++ b/1_8_R3/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_8_R3</artifactId>

--- a/1_9_R1/pom.xml
+++ b/1_9_R1/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_9_R1</artifactId>

--- a/1_9_R2/pom.xml
+++ b/1_9_R2/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-1_9_R2</artifactId>

--- a/README.md
+++ b/README.md
@@ -19,7 +19,7 @@ AnvilGUI requires the usage of Maven or a Maven compatible build system.
 <dependency>
     <groupId>net.wesjd</groupId>
     <artifactId>anvilgui</artifactId>
-    <version>1.5.2-SNAPSHOT</version>
+    <version>1.5.3-SNAPSHOT</version>
 </dependency>
 
 <repository>

--- a/abstraction/pom.xml
+++ b/abstraction/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui-abstraction</artifactId>

--- a/api/pom.xml
+++ b/api/pom.xml
@@ -7,7 +7,7 @@
     <parent>
         <groupId>net.wesjd</groupId>
         <artifactId>anvilgui-parent</artifactId>
-        <version>1.5.2-SNAPSHOT</version>
+        <version>1.5.3-SNAPSHOT</version>
     </parent>
 
     <artifactId>anvilgui</artifactId>

--- a/api/src/main/java/net/wesjd/anvilgui/version/VersionMatcher.java
+++ b/api/src/main/java/net/wesjd/anvilgui/version/VersionMatcher.java
@@ -2,9 +2,6 @@ package net.wesjd.anvilgui.version;
 
 import org.bukkit.Bukkit;
 
-import java.util.Arrays;
-import java.util.List;
-
 /**
  * Matches the server's NMS version to its {@link VersionWrapper}
  *
@@ -14,23 +11,19 @@ import java.util.List;
 public class VersionMatcher {
 
     /**
-     * The server's version
-     */
-    private final String serverVersion = Bukkit.getServer().getClass().getPackage().getName().split("\\.")[3].substring(1);
-
-    /**
      * Matches the server version to it's {@link VersionWrapper}
      *
      * @return The {@link VersionWrapper} for this server
-     * @throws RuntimeException If AnvilGUI doesn't support this server version
+     * @throws IllegalStateException If the version wrapper failed to be instantiated or is unable to be found
      */
     public VersionWrapper match() {
+        final String serverVersion = Bukkit.getServer().getClass().getPackage().getName().split("\\.")[3].substring(1);
         try {
-            return (VersionWrapper) Class.forName(this.getClass().getPackage().getName() + ".Wrapper" + serverVersion).newInstance();
-        } catch (IllegalAccessException | InstantiationException ex) {
-            throw new RuntimeException(ex);
-        } catch (ClassNotFoundException e) {
-            throw new RuntimeException("Your server version isn't supported in AnvilGUI!");
+            return (VersionWrapper) Class.forName(getClass().getPackage().getName() + ".Wrapper" + serverVersion).newInstance();
+        } catch (IllegalAccessException | InstantiationException exception) {
+            throw new IllegalStateException("Failed to instantiate version wrapper for version " + serverVersion, exception);
+        } catch (ClassNotFoundException exception) {
+            throw new IllegalStateException("AnvilGUI does not support server version \"" + serverVersion + "\"", exception);
         }
     }
 

--- a/api/src/main/java/net/wesjd/anvilgui/version/VersionMatcher.java
+++ b/api/src/main/java/net/wesjd/anvilgui/version/VersionMatcher.java
@@ -17,28 +17,6 @@ public class VersionMatcher {
      * The server's version
      */
     private final String serverVersion = Bukkit.getServer().getClass().getPackage().getName().split("\\.")[3].substring(1);
-    /**
-     * All available {@link VersionWrapper}s
-     */
-    private final List<Class<? extends VersionWrapper>> versions = Arrays.asList(
-            Wrapper1_7_R4.class,
-            Wrapper1_8_R1.class,
-            Wrapper1_8_R2.class,
-            Wrapper1_8_R3.class,
-            Wrapper1_9_R1.class,
-            Wrapper1_9_R2.class,
-            Wrapper1_10_R1.class,
-            Wrapper1_11_R1.class,
-            Wrapper1_12_R1.class,
-            Wrapper1_13_R1.class,
-            Wrapper1_13_R2.class,
-            Wrapper1_14_R1.class,
-            Wrapper1_15_R1.class,
-            Wrapper1_16_R1.class,
-            Wrapper1_16_R2.class,
-            Wrapper1_16_R3.class,
-            Wrapper1_17_R1.class
-    );
 
     /**
      * Matches the server version to it's {@link VersionWrapper}
@@ -48,12 +26,11 @@ public class VersionMatcher {
      */
     public VersionWrapper match() {
         try {
-            return versions.stream()
-                    .filter(version -> version.getSimpleName().substring(7).equals(serverVersion))
-                    .findFirst().orElseThrow(() -> new RuntimeException("Your server version isn't supported in AnvilGUI!"))
-                    .newInstance();
+            return (VersionWrapper) Class.forName(this.getClass().getPackage().getName() + ".Wrapper" + serverVersion).newInstance();
         } catch (IllegalAccessException | InstantiationException ex) {
             throw new RuntimeException(ex);
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException("Your server version isn't supported in AnvilGUI!");
         }
     }
 

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 
     <groupId>net.wesjd</groupId>
     <artifactId>anvilgui-parent</artifactId>
-    <version>1.5.2-SNAPSHOT</version>
+    <version>1.5.3-SNAPSHOT</version>
     <packaging>pom</packaging>
 
     <properties>

